lib/ansible/modules/network/ios/ios_config.py

```diff
@@ -434,7 +452,11 @@ def main():
         contents = get_config(module, flags=flags)
         config = NetworkConfig(indent=1, contents=contents)
         if module.params['backup']:
-            result['__backup__'] = contents
+            if module.params['remove_extraneous']:
+                contents = re.sub(REMOVE_EXTRANEOUS_DEFAULT, '', contents)
```

Comment:
Here 'contents' has been overwritten with configs without extra lines. so later in this module this new content would be used to generate diff. This might cause problem in case someone is playing stored config from device with extra lines (use-case is where someone updated ansible with this fix and trying to play diffs stored without this fix). Better to avoid this overwrite.

Reply:
I might think it then might be better to strip the lines from stored config, then. I haven't looked closely at it, but the device responds with periods to indicate that some work is taking place, are there times when it is a number other than three? If this prefix can vary for reasons not related to device configuration, it would seem to make sense to strip it on all configuration.

Reply:
@gdpak @Qalthos I have updated the code based on the earlier discussion of implementing via `filter_lines` option which gives more flexibility to the user and has discussed the scenario with @IPvSean as well and he's also in agreement with the respective idea. Also, now I am not updating the `contents` variable altogether which resolves the @gdpak review comment. 
Kindly review the updated PR.